lib/spack/spack/compilers/__init__.py

```diff
@@ -118,13 +118,104 @@ def get_compiler_config(scope=None, init_config=True):
 
     merged_config = spack.config.get("compilers")
     if merged_config:
+        # Config is empty for this scope
+        # Do not init config because there is a non-empty scope
         return config
 
     _init_compiler_config(scope=scope)
     config = spack.config.get("compilers", scope=scope)
     return config
 
 
+def get_compiler_config_from_packages(scope=None):
+    """Return the compiler configuration from packages.yaml"""
+    config = spack.config.get("packages", scope=scope)
+    if not config:
+        return []
+
+    packages = []
+    compiler_package_names = supported_compilers() + list(package_name_to_compiler_name.keys())
```

Comment:
(nitpick, we'll eventually change this one in another PR):
```suggestion
    compiler_package_names = supported_compilers() + list(package_name_to_compiler_name)
```